src/components/editor/editor-screen-table.tsx

```diff
@@ -0,0 +1,374 @@
+      positiveButtonText={ strings.generic.confirm }
+      title={ strings.editorScreen.confirmDeleteVersionTitle }
+      text={ strings.editorScreen.confirmDeleteVersionText }
+      open={ deleteDialogOpen }
+    />
+  );
+  
+  /**
+   * Renders version menu
+   */
+  const renderVersionMenu = () => (
+    <Popover
+      open={ popoverOpen }
+      anchorEl={ popoverAnchorElement }
+      onClose={ handleMenuClose }
+      anchorOrigin={{
+        vertical: "bottom",
+        horizontal: "right"
+      }}
+      transformOrigin={{
+        vertical: "top",
+        horizontal: "right"
+      }}
+    >
+      <List sx={{ p: theme.spacing(0.5) }}>
+        <ListItem sx={{ p: theme.spacing(0.5) }}>
+          <ListItemButton
+            sx={{ p: 0 }}
+            onClick={ () => setDeleteDialogOpen(!deleteDialogOpen) }
+          >
+            <ListItemIcon sx={{ minWidth: 0, mr: theme.spacing(1) }}>
+              <Delete/>
+            </ListItemIcon>
+            <ListItemText primary={ strings.generic.delete }/>
+          </ListItemButton>
+        </ListItem>
+        <ListItem sx={{ p: theme.spacing(0.5) }}>
+          <ListItemButton
+            sx={{ p: 0 }}
+            component={ Link }
+            to={ linkToEditor ?? "#" }
+          >
+            <ListItemIcon sx={{ minWidth: 0, mr: theme.spacing(1) }}>
+              <Settings/>
+            </ListItemIcon>
+            <ListItemText primary={ strings.generic.edit }/>
+          </ListItemButton>
+        </ListItem>
+      </List>
+    </Popover>
+  );
+
+  /**
+   * Builds MetaformVersionRow for production version of Metaform
+   * 
+   * @param id metaform id
+   */
+  const buildProductionVersion = (id: string): MetaformVersionRow => {
+    return {
+      id: id,
+      typeString: strings.editorScreen.formProductionVersion
+    };
+  };
+
+  /**
+   * Builds MetaformVersionRows
+   * 
+   * @param metaform metaform
+   */
+  const buildVersionRows = (metaform: Metaform): MetaformVersionRow[] => {
+    const versions = metaformVersions.filter(x => x.data.id as unknown === metaform.id);
+    const productionVersion = buildProductionVersion(metaform.id!);
+    const versionRows: MetaformVersionRow[] = versions.map((version: MetaformVersion) => {
+      return {
+        id: version.id!,
+        typeString: version.type === MetaformVersionType.Archived ? strings.editorScreen.formVersionArchived : strings.editorScreen.formVersionDraft,
+        type: version.type,
+        createdAt: version.createdAt!,
+        modifiedAt: version.modifiedAt!,
+        modifierId: version.lastModifierId!
+      };
+    });
+    versionRows.push(productionVersion);
+
+    return versionRows;
+  };
+
+  useEffect(() => {
+    /**
+     * Handles correct link to editor for edit button
+     */
+    const handleEditorLink = async () => !!selectedId && setLinkToEditor(await getPathToEditor(selectedId));
+
+    if (selectedId) {
+      handleEditorLink();
+    }
```

Comment:
Move logic to its own method and run only that inside useEffect

Reply:
Done :)